### PR TITLE
Google Drive integration

### LINKED ISSUE #408: Google Drive to S3 sync for videos
Part of https://github.com/mitodl/ocw-studio/issues/403

- [ ] Service account for Google Drive
- [ ] Shared folder for the service account to store videos
- [ ] Set up a watch on the folder, with recurring task to update the expiration
- [ ] Set up an endpoint for push notifications from GDrive, to upload new/modified files to S3

### PR DESCRIPTION
**Pull request by mbertrand**
#### Pre-Flight checklist

- [x] Migrations
  - [x] Migration is backwards-compatible with current production code
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested
- [x] Settings
  - [x] New settings are documented and present in `app.json`
  - [x] New settings have reasonable development defaults, if applicable
  - [x] Opened issue for DevOps regarding necessary configuration changes to deployed environments

#### What are the relevant tickets?
Closes #408

#### What's this PR do?
Adds tasks, models and functions to check for new video files in a particular google team drive, and if a matching website is found based on the parent folder(s), upload them to S3.

#### How should this be manually tested?
- Set the following in your `.env` file:
  ```
  DRIVE_SERVICE_ACCOUNT_CREDS=<ask me>
  DRIVE_SHARED_ID=<ask me>
  AWS_STORAGE_BUCKET_NAME* = <same as on RC, ditto for AWS keys>
  ```
- Create or choose an existing `Website`, note the `short_id`, and create a folder in the team drive with the `short_id` as the name.  Create a subfolder under that.  Then upload a video file to that subfolder.
- Wait.  Within a few minutes, a celery task should run, create a `DriveFile` object for that video, and upload the video to S3.

#### Where should the reviewer start?
#403 

#### Any background context you want to provide?
The original plan was to create a Google Drive push notification that would trigger whenever a file was added/modified, and add a webhook endpoint in studio to process that.  But there seems to be a bug with team drives that prevents the notification from being triggered unless the file is added/modified by the service account that created the push notification.

### CODE REVIEW

**Comment by pdpinch** [comment]
Does this close #408 ?

**Review by abeglova — APPROVED** [review]
Works great! LGTM

File: gdrive_sync/api.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    return tree[1:]  # first one is the drive
+
+
+def process_file_result(file_obj: Dict):
+    """Convert an API file response into DriveFile objects"""
+    parents = file_obj.get("parents")
+    if parents:
+        folder_tree = get_parent_tree(parents)
+        for folder in folder_tree:
+            print(folder["name"])
```
Comment (abeglova) [inline gdrive_sync/api.py:91]: Just noticed you forgot to remove a print()

**Comment by mbertrand** [comment]
@abeglova would you mind taking another look? I made changes to avoid repeated S3 uploads if the checksums are identical.  Also made changes to the initial migration so you'll need to run `manage.py migrate gdrive_sync zero` followed by `manage.py migrate gdrive_sync`

**Review by abeglova — APPROVED** [review]
Works as advertised. LGTM

**Pull request merged**